dvc/repo/experiments/__init__.py

```diff
@@ -362,6 +387,10 @@ def _update_params(self, params: dict):
             suffix = path.suffix.lower()
             modify_data = MODIFIERS[suffix]
             with modify_data(path, fs=self.repo.fs) as data:
+                new_params = self._get_new_params(data, params[params_fname])
+                if new_params:
+                    error_msg = self._format_new_params_msg(new_params, path)
+                    raise MissingParamsError(error_msg)
```

Comment:
@mattlbeck, could you please move this to `merge_params`? That way, you can just test it via unittest, though you may need to adjust existing ones. Although, we could extend `merge_params`'s API with `only_update=True|False`.